crates/bevy_app/src/app_builder.rs

```diff
@@ -534,17 +580,19 @@ impl AppBuilder {
         self
     }
 
-    /// Registers a new component using the given [ComponentDescriptor]. Components do not need to
-    /// be manually registered. This just provides a way to override default configuration.
-    /// Attempting to register a component with a type that has already been used by [World]
-    /// will result in an error.
+    /// Registers a new component using the given [ComponentDescriptor].
+    ///
+    /// Components do not need to be manually registered. This just provides a way to
+    /// override default configuration. Attempting to register a component with a type
+    /// that has already been used by [World] will result in an error.
     ///
     /// See [World::register_component]
     pub fn register_component(&mut self, descriptor: ComponentDescriptor) -> &mut Self {
         self.world_mut().register_component(descriptor).unwrap();
         self
     }
 
+    /// Adds the type `T` to the type registry resource.
```

Comment:
An explanation of why you might want to do so would be appreciated.

IIRC it's used to enable reflection?

Reply:
From a quick "find references" and git blame it looks like it's primarily, if not solely, used for reflection. I don't feel extremely confident about what to add since I never used reflection directly.